libp2p/messagesHandler_test.go

```diff
@@ -0,0 +1,1213 @@
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.OutgoingCLB = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilChannelLoadBalancer, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("nil Marshaller should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.Marshaller = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilMarshaller, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("nil ConnMonitorWrapper should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.ConnMonitorWrapper = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilConnectionMonitorWrapper, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("nil PeersRatingHandler should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.PeersRatingHandler = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilPeersRatingHandler, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("nil Debugger should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.Debugger = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilDebugger, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("nil SyncTimer should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.SyncTimer = nil
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, p2p.ErrNilSyncTimer, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("RegisterMessageHandler fails", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgMessagesHandler()
+		args.DirectSender = &mock.DirectSenderStub{
+			RegisterDirectMessageProcessorCalled: func(handler p2p.MessageProcessor) error {
+				return expectedError
+			},
+		}
+		mh, err := libp2p.NewMessagesHandler(args)
+		assert.Equal(t, expectedError, err)
+		assert.True(t, check.IfNil(mh))
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		mh, err := libp2p.NewMessagesHandler(createMockArgMessagesHandler())
+		assert.Nil(t, err)
+		assert.False(t, check.IfNil(mh))
+		assert.Nil(t, mh.Close())
+	})
+	t.Run("process loop", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("CollectOneElementFromChannels always returns nil", func(t *testing.T) {
+			t.Parallel()
+
+			args := createMockArgMessagesHandler()
+			args.OutgoingCLB = &mock.ChannelLoadBalancerStub{
+				CollectOneElementFromChannelsCalled: func() *libp2p.SendableData {
+					return nil
+				},
+			}
+			args.Marshaller = &mock.MarshallerStub{
+				MarshalCalled: func(obj interface{}) ([]byte, error) {
+					assert.Fail(t, "should not have been called")
+					return nil, nil
+				},
+			}
+			mh, _ := libp2p.NewMessagesHandler(args)
+			assert.False(t, check.IfNil(mh))
+			time.Sleep(time.Millisecond * 3)
```

Comment:
suggestion to increase to at least 50ms all these sleep calls in order to avoid false positives caused by low spec machines

Reply:
changed